packages/frontend-shared/README.md

```diff
@@ -18,26 +22,143 @@ For the best development experience, you will want to use VS Code with the [Vola
+#### [Pinia](https://pinia.vuejs.org/)
+TS-friendly, modularized state management - this is what we use instead of Vuex for the small amount of global state we have.
+
+#### [VueUse](https://vueuse.org/)
+Broad collection of composable utilities that provides reactive values for various common events and DOM properties, CSS rules, local storage and a lot of other stuff. Very useful if we feel the need to add and tear down low level event listeners, as this library already covers so many and exposes them to Vue correctly. It's like a front-end lodash, so where the VueUse implementation of something works for us, we prefer to use it instead of roll our own.
+
+#### [Headless UI](https://headlessui.dev/)
+We are using some components from Headless UI as the basis for UI patterns like modals, custom dropdowns, and expanding panels. We use Headless UI because it is well documented and the accessibility features are properly thought out. These advantages outweigh the occasional workarounds we have to use in order to get sophisticated behavior working that Headless UI does not support.
+
+### Router
+
+Only `@packages/app` has a router, so details are described in its [README](../app/README.md).
+### Styles
+
+#### Tailwind and Windi
+We use [Tailwind](https://tailwindcss.com/) through [WindiCSS](https://windicss.org/). The codebase is utility-driven and all CSS that can be achieved through utility classes is written that way. The main way to reuse CSS in multiple places is to extract a component that applies the utility classes and can wrap other elements as needed.
+
+#### Explicit Pixel Values
+WindiCSS can create CSS classes as build time based on what class names we use in our components. That means syntax like this will work:
+
+`<p class="p-20px">`
+
+This allows us to specify explicit pixel values for measurements. We follow this pattern throughout the Cypress App codebase.
+
+### Icons
+
+The icons will temporarily live in this package and will soon be moved to `packages/frontend-shared`.
+
+## Custom Icon Library
+Cy has a very custom icon library, to meet the following needs:
+
+* Most of our icons are duo-tone
+* They must be styled in both dark and light contexts (e.g. on a dark menu bar vs on a light background).
+* Since they're duotone, you want to target the specific strokes and fills of the SVGs to color them.
+* All of the niceties of our utility classes should work (e.g. `group-hover` or `group-focus`).
+
+### Usage
+#### Importing Icons
+1. `import MyIcon from '~icons/cy/path-to-icon_x16'`
+2. Automatic template discovery
+```jsx
+/* This just works. No imports necessary */
+<i-cy-path-to-icon_x16 />
+```
+
+#### Styling Icons
+```jsx
+/* This renders a book icon from `./src/assets/book_x16.svg`
+ * and makes it pink and purple
+ */
+
+<i-cy-book_x16 class="
```

Comment:
Does this example support both dark and light? I see only dark classes here - if something is working automagically, that'd be nice to have explained.

Reply:
I'll add image. `dark` and `light` really refer to like, the primary and secondary colors present in the icon, so this example supports setting both colors for the icon, and setting those colors to be changed in the hover state. Using the `hover:` psuedoclass is one of the niceties mentioned earlier.